pypuppetdb/api.py

```diff
@@ -398,35 +398,31 @@ def nodes(self, unreported=2, with_status=False, **kwargs):
             nodes = [nodes, ]
 
         if with_status:
-            latest_events = self.event_counts(
-                query=EqualsOperator("latest_report?", True),
-                summarize_by='certname'
-            )
+            reports = {}
+            list_query = OrOperator()
+
+            def _query_reports():
+                query = AndOperator()
+                query.add(list_query)
+                query.add(EqualsOperator("latest_report?", True))
+                for report in self.reports(query=query):
+                    reports[report.node] = report
+
+            for node in nodes:
+                list_query.add(EqualsOperator('certname', node['certname']))
+                # Fixed string size set to 4K - Avoid RequestTooLong error
+                if len(str(list_query)) > 4096:
```

Comment:
I'm not sure on this one.  When the query > 4096 it should be first to be a POST and not a GET.

I was doing some testing and found even though I requested a post the &query= was still on the request.